pkg/apis/options/header.go

```diff
@@ -0,0 +1,44 @@
+package options
+
+// Header represents an individual header that will be added to a request or
+// response header.
+type Header struct {
+	// Name is the header name to be used for this set of values.
+	// Names should be unique within a list of Headers.
+	Name string `json:"name"`
+
+	// PreserveRequestValue determines whether any values for this header
+	// should be preserved for the request to the upstream server.
+	// This option only takes effet on injected request headers.
+	// Defaults to false (headers that match this header will be stripped).
+	PreserveRequestValue bool `json:"preserveRequestValue"`
+
+	// Values contains the desired values for this header
+	Values []HeaderValue `json:"values"`
+}
+
+// HeaderValue represents a single header value and the sources that can
+// make up the header value
+type HeaderValue struct {
+	// Allow users to load the value from a secret source
+	*SecretSource
+
+	// Allow users to load the value from a session claim
+	*ClaimSource
+}
+
+// ClaimSource allows loading a header value from a claim within the session
+type ClaimSource struct {
+	// Claim is the name of the claim in the session that the value should be
+	// loaded from.
+	Claim string `json:claim,omitempty`
+
+	// Prefix is an optional prefix that will be prepended to the value of the
+	// claim if it is non-empty.
+	Prefix string `json:prefix,omitempty`
+
+	// BasicAuthPassword converts this claim into a basic auth header.
+	// Note the value of claim will become the basic auth username and the
+	// basicAuthPassword will be used as the password value.
+	BasicAuthPassword *SecretSource
```

Comment:
I think its worth additionally allowing stuff to be put into the authorization header -- since we've had users in the past want the access token put there instead of the ID Token (because X-Forwarded-Access-Token didn't work for them).

Reply:
Scratch that -- I see the purpose of this (to allow base64 + appending password) where Prefix could be used to create 'Bearer ' + claim

Reply:
In that case does it make sense to consolidate Prefix + BasicAuthPassword into a `transformer` func handler that is more generic (that could do that logic, or whatever else advanced we need)?

Reply:
Could potentially make it a transformer func, but is that more of an implementation detail in the handlers rather than a config question? Or have you something broader in mind for tranforms?

Reply:
Yeah, I see your point.  Matching these to the config -> implementation path explicitly keeps things aligned and clean.